rerun_py/rerun_sdk/rerun/_rerun2/datatypes/quaternion_ext.py

```diff
@@ -0,0 +1,10 @@
+from __future__ import annotations
+
+from typing import Any
+
+import numpy.typing as npt
+
+
+class QuaternionExt:
+    def __init__(self: Any, *, xyzw: npt.ArrayLike) -> None:
```

Comment:
this one inherited a bug I fixed here, plz incorporate :)
https://github.com/rerun-io/rerun/commit/0284d03f123b23ee8586f1d76764643bdc34efbb

Reply:
Let's discuss this one tomorrow -- going to merge as is because I think this version is more correct.

Reply:
The thing I ran into was that the callsite was incorrect for this https://github.com/rerun-io/rerun/blob/2bf7d8c1b187678c6613a8efb6b6aa9e611b5884/rerun_py/rerun_sdk/rerun/_rerun2/datatypes/quaternion.py#L36
i.e. the __init__ function of Quaternion was actually broken. But with your changes that’s no longer a concern actually